lib/XMLHttpRequest.js

```diff
@@ -46,6 +72,12 @@ function XMLHttpRequest(opts) {
   var http = require('http');
   var https = require('https');
 
+  var max_redirects = opts.maxRedirects;
+  if (max_redirects == null || isNaN(max_redirects)) max_redirects = 20;
```

Comment:
I'd really like to stay away from having non-strict equality operators. There's lots of hidden quirks (like `[] == false`) that are easily overlooked and the syntax isn't immediately obvious.
For the same reason, we should be using `Number.isNaN` as just `isNaN` will also check if the value can be coerced to a number (which is probably not what you want).

Looking at your case I'd replace this with:
```suggestion
  if (typeof max_redirects !== 'number' || Number.isNaN(max_redirects)) max_redirects = 20;
```

Reply:
It's a really good approach, thanks for this!